test/integration/testdata/main.py

```diff
@@ -25,13 +25,13 @@ def check_env_var_handler(event, context):
 
 def assert_env_var_is_overwritten(event, context):
     print(os.environ.get("AWS_LAMBDA_FUNCTION_NAME"))
-    if os.environ.get("AWS_LAMBDA_FUNCTION_NAME") == "test_function":
+    if os.environ.get("AWS_LAMBDA_FUNCTION_NAME") == "function":
         raise("Function name was not overwritten")
     else:
         return "My lambda ran succesfully"
 
 def assert_lambda_arn_in_context(event, context):
-    if context.invoked_function_arn == f"arn:aws:lambda:us-east-1:012345678912:function:{os.environ.get('AWS_LAMBDA_FUNCTION_NAME', 'test_function')}":
+    if context.invoked_function_arn == f"arn:aws:lambda:us-east-1:012345678912:function:{os.environ.get('AWS_LAMBDA_FUNCTION_NAME', 'function')}":
```

Comment:
I think I found an issue with our tests thanks to this change. 
Because the integration tests are passing but here they should fail.

Your new test works correctly with this code, but there was already a previous test `test_lambda_function_arn_exists()` -> `arnexists-x86_64` that doesn't pass the new env var, so this ARN should have `test_function` instead (the old value).

Looking at the [output of the test run](https://github.com/aws/aws-lambda-runtime-interface-emulator/actions/runs/9861242832/job/27231798350?pr=46) `arnexists-x86_64` doesn't appear on the list of tests that were run, so it looks it might be ignoring the failing tests instead of fail overall.

I'll have to figure out this issue, but in the meantime you should change this to have different cases depending if the new env var was passed or not (probably just create a new handler here to be called form the new integ test).

The previous handler in this file also has the same issue, but because it's the branch that shouldn't happen, then it's not relevant and the test still succeed, but you should change that too.


Everything else looks good

Reply:
Digging a little deeper, it looks like this shouldn't really be a problem, because when running this we will always have a value for `AWS_LAMBDA_FUNCTION_NAME`, so in this code it will never go to the second parameter of the `os.environ.get()`, 

It's still weird that the previous test that uses this code doesn't show up on the output for the tests, so I'm still investigating what's going on there. 

Did you make this change because the tests failed if you didn't? Or did you change it before actually running the tests? (My point is that: either this change is making the old test fail, or this change shouldn't be needed).

Reply:
Of course it had to be a simple thing. The old test is  not running because the new test you added has the same name! (`def test_lambda_function_arn_exists`) so it's actually overwriting the old test, but it succeeds. 


What I would say is: don't change this file. This change doesn't actually affect the test because the env var is always set. We could probably just remove that second parameter in the test, but for now, it's better if you don't modify the file, so it doesn't get mixed up as a needed change.